runtime/vm/CRIUHelpers.cpp

```diff
@@ -263,6 +266,60 @@ criuRestoreInitializeTrace(J9VMThread *currentThread, void *userData, const char
 	return result;
 }
 
+/**
+ * An internal JVM checkpoint hook to initialize -Xrs after restore. This only
+ * supports -Xrs options with "onRestore" appended to it.
+ *
+ * @param[in] currentThread vmThread token
+ * @param[in] userData J9InternalHookRecord pointer
+ * @param[in/out] nlsMsgFormat an NLS message
+ *
+ * #return Always returns TRUE
+ */
+static BOOLEAN
+criuRestoreInitializeXrs(J9VMThread *currentThread, void *userData, const char **nlsMsgFormat)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	IDATA argIndex = 0;
+	J9VMInitArgs *args = vm->checkpointState.restoreArgsList;
+
+	if (NULL != args) {
+		if ((argIndex = FIND_ARG_IN_ARGS(args, OPTIONAL_LIST_MATCH, VMOPT_XRS, NULL)) >= 0) {
+			bool argProcessed = false;
+			char* optionValue = NULL;
+			U_32 sigOptions = 0;
+
+			GET_OPTION_VALUE_ARGS(args, argIndex, ':', &optionValue);
+			if (NULL != optionValue) {
+				if (0 == strcmp(optionValue, "syncOnRestore")) {
```

Comment:
> and avoid adding new option strings?

We can't use the same options as the behaviour is not identical. This would require to update documentation for existing options which is less desired. 

> In addition, should we handle multiple incoming -Xrs in the options file?

The current behaviour matches what we do on startup